pretyping/globEnv.ml

```diff
@@ -97,11 +97,19 @@ let push_rec_types ~hypnaming sigma (lna,typarray) env =
   let env,ctx = Array.fold_left_map (fun e assum -> let (d,e) = push_rel sigma assum e ~hypnaming in (e,d)) env ctxt in
   Array.map get_annot ctx, env
 
-let new_evar env sigma ?src ?naming typ =
+let new_evar env sigma ?src ?(naming = Namegen.IntroAnonymous) typ =
   let (subst, _, sign) as ext = Lazy.force env.extra in
   let instance = Evarutil.default_ext_instance ext in
   let typ' = csubst_subst sigma subst typ in
-  let (sigma, evk) = new_pure_evar sign sigma typ' ?src ?naming in
+  let name = match naming with
+  | IntroAnonymous -> None
+  | IntroIdentifier id -> Some id
+  | IntroFresh id ->
+    let has_name id = try let _ = Evd.evar_key id sigma in true with Not_found -> false in
+    let id = Namegen.next_ident_away_from id has_name in
+    Some id
```

Comment:
I don't like having to duplicate this code, could we have it as its own function in namegen or evarutil?